src/com/hms_networks/americas/sc/realtimedata/RealTimeDataQueueManager.java

```diff
@@ -0,0 +1,197 @@
+import com.hms_networks.americas.sc.logging.Logger;
+import com.hms_networks.americas.sc.taginfo.TagGroup;
+import com.hms_networks.americas.sc.taginfo.TagInfo;
+import com.hms_networks.americas.sc.taginfo.TagInfoManager;
+import com.hms_networks.americas.sc.taginfo.TagType;
+
+public class RealTimeDataQueueManager {
+
+  // hold the tag managers in array
+
+  /** Holds an array of lists of data points for every tag in group */
+  private static ArrayList tagManagers;
+
+  /** Holds an array of the lists of what tags are in each group */
+  private static ArrayList tagGroupList;
+
+  /**
+   * lists of iterator for each tag group used to get a single point at a time from a tag group
+   * while retrieving data points.
+   */
+  private static int[] tagManagerIterators;
+
+  /** Index of information for tag group A in class ArrayLists. */
+  public static final int GROUP_A = 0;
+  /** Index of information for tag group B in class ArrayLists. */
+  public static final int GROUP_B = 1;
+  /** Index of information for tag group C in class ArrayLists. */
+  public static final int GROUP_C = 2;
+  /** Index of information for tag group D in class ArrayLists. */
+  public static final int GROUP_D = 3;
+
+  /** The number of tag groups. */
+  private static final int NUM_TAG_GROUPS = 4;
+
+  /** Store the tag groups in an ArrayList for access by tag group index. */
+  private static void initTagGroups() {
+
+    tagGroupList.add(TagInfoManager.getTagInfoListFiltered(TagGroup.A));
+    tagGroupList.add(TagInfoManager.getTagInfoListFiltered(TagGroup.B));
+    tagGroupList.add(TagInfoManager.getTagInfoListFiltered(TagGroup.C));
+    tagGroupList.add(TagInfoManager.getTagInfoListFiltered(TagGroup.D));
+  }
+
+  /**
+   * Initialize a tag manager object for each tag in the tag's group.
+   *
+   * @throws EWException If the TagControl object fails to initialize for a given tag.
+   */
+  private static void initTagManagers() throws EWException {
+
+    tagManagers = new ArrayList();
+
+    for (int tagGroupIndex = 0; tagGroupIndex < NUM_TAG_GROUPS; tagGroupIndex++) {
+      ArrayList tagManagerTmp = new ArrayList();
+      for (int tagListIndex = 0;
+          tagGroupIndex < ((ArrayList) tagGroupList.get(tagGroupIndex)).size();
+          tagGroupIndex++) {
+        TagInfo currentTag =
+            ((TagInfo) ((ArrayList) tagGroupList.get(tagListIndex)).get(tagGroupIndex));
+        tagManagerTmp.add(new RealTimeTagDataPointManager(currentTag.getName()));
+      }
+      tagManagers.add(tagManagerTmp);
+    }
+  }
+
+  /**
+   * Tag iterators are used to retrieve a single data point from each tag before going to the next
+   * tag in a tag group list. Initialize all tag iterators to 0 to start at the beginning of the
+   * lists.
+   */
+  private static void initTagManageIterators() {
+
+    tagManagerIterators = new int[NUM_TAG_GROUPS];
+
+    for (int i = 0; i < NUM_TAG_GROUPS; i++) {
+      tagManagerIterators[i] = 0;
+    }
+  }

```

Comment:
```suggestion
  private static void initTagManageIterators() {
    tagManagerIterators = new int[NUM_TAG_GROUPS];
    for (int i = 0; i < NUM_TAG_GROUPS; i++) {
      tagManagerIterators[i] = 0;
    }
  }
```
Spacing in the method is inconsistent with many other methods.